dcplib/checksumming_io/checksumming_buffered_reader.py

```diff
@@ -3,14 +3,25 @@
 from io import BufferedReader
 from .s3_etag import S3Etag
 
+from dcplib import s3_multipart
+
 
 class ChecksummingBufferedReader:
 
     def __init__(self, *args, **kwargs):
+        """
+        :param read_file_size: optional file size for correctly setting the s3 etag chunk size
```

Comment:
so if this is optional, what you probably want to do is to track the total number of bytes sent to `ChecksummingBufferedReader`.  if it would have warranted a larger chunk size, it should throw an exception.

Reply:
In what case would it merit a larger chunk size?

Reply:
At the end, if the chunk size is unspecified, you could call `get_s3_multipart_chunk_size(..)` with the number of bytes that got processed.  If that is not `S3Etag.default_chunk_size`, then the checksum calculated is bad.

Or you could just make the parameter required. :)

Reply:
Making it required. This might be a pain in some cases, but it will make it easier to understand.